### PR TITLE
NATIVE-608: Fixed Windows' Yarn-Path error by using OS conditions

### PR DESCRIPTION
**Pull request by lulo4907**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **NATIVE**.

### CODE REVIEW

**Review by maxammann — APPROVED** [review]

File: android/app/buildConfigs.gradle [hunk 1]
```diff
@@ -12,7 +12,12 @@ def determineBuildConfigName() {
 def createBuildConfig(buildConfigName) {
     logger.quiet("Using build config $buildConfigName")
 
-    def command = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+    def command
+    if (System.properties['os.name'].toLowerCase().contains('windows')) {
+        command = ["cmd", "/c", "yarn --silent babel-node tools/build-config to-json $buildConfigName"]
+    } else {
+        command = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+    }
```
Comment (klinzo) [inline android/app/buildConfigs.gradle:20]:
```suggestion
    def command
    def baseCommand = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
    if (System.properties['os.name'].toLowerCase().contains('windows')) {
        command = ["cmd", "/c", baseCommand]
    } else {
        command = baseCommand
    }
```

**Review by klinzo — CHANGES_REQUESTED** [review]

File: android/app/buildConfigs.gradle [hunk 2]
```diff
@@ -13,10 +17,11 @@ def createBuildConfig(buildConfigName) {
...
-        command = ["cmd", "/c", "yarn --silent babel-node tools/build-config to-json $buildConfigName"]
+    def baseCommand = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+    if (isWindows() == "true") {
+        command = ["cmd", "/c", baseCommand]
     } else {
-        command = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+        command = baseCommand
     }
     def proc = command.execute()
     proc.waitFor()
```
Comment (klinzo) [inline android/app/buildConfigs.gradle]: This (and 4 more lines) is exactly the same as below, right? Could you maybe move it to a separate function?

**Review by klinzo — APPROVED** [review]

File: android/app/buildConfigs.gradle [hunk 3]
```diff
@@ -1,5 +1,9 @@
 import groovy.json.JsonSlurper
 
+def isWindows() {
+    System.properties['os.name'].toLowerCase().contains('windows') ? "true" : "false"
```
Comment (maxammann) [inline android/app/buildConfigs.gradle:4]:
```suggestion
    System.properties['os.name'].toLowerCase().contains('windows')
```

File: android/app/buildConfigs.gradle [hunk 4]
```diff
@@ -12,7 +16,13 @@ def determineBuildConfigName() {
 def createBuildConfig(buildConfigName) {
     logger.quiet("Using build config $buildConfigName")
 
-    def command = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+    def command
+    def baseCommand = "yarn --silent babel-node tools/build-config to-json $buildConfigName"
+    if (isWindows() == "true") {
```
Comment (maxammann) [inline android/app/buildConfigs.gradle:21]:
```suggestion
    if (isWindows()) {
```

File: android/app/buildConfigs.gradle [hunk 5]
```diff
@@ -48,7 +58,13 @@ def setupGoogleServices(buildConfig, resValue) {
 }
 
 def setupResourceValues(buildConfigName, resValue) {
-    def command = "yarn --silent babel-node tools/build-config to-xcconfig $buildConfigName"
+    def command
+    def baseCommand = "yarn --silent babel-node tools/build-config to-xcconfig $buildConfigName"
```
Comment (maxammann) [inline android/app/buildConfigs.gradle:62]: This code can be deduplicated by moving it to a separate function: executeProcess(command) which returns the output as string

**Review by maxammann — CHANGES_REQUESTED** [review]

File: android/app/buildConfigs.gradle [hunk 6]
```diff
@@ -13,17 +17,12 @@ def determineBuildConfigName() {
     }
 }
 
+
```
Comment (klinzo) [inline android/app/buildConfigs.gradle:20]: unnecessary line

**Review by klinzo — APPROVED** [review]

File: android/app/buildConfigs.gradle [hunk 7]
```diff
@@ -1,7 +1,11 @@
 import groovy.json.JsonSlurper
 
 def isWindows() {
-    System.properties['os.name'].toLowerCase().contains('windows') ? "true" : "false"
+    System.properties['os.name'].toLowerCase().contains('windows')
+}
+
+def execCommand(command) {
+    isWindows() ? ["cmd", "/c", command] : command
```
Comment (maxammann) [inline android/app/buildConfigs.gradle:8]: This function does not execute anything. Add the `.execute()` here.

**Review by maxammann — CHANGES_REQUESTED** [review]

File: android/app/buildConfigs.gradle (hunk 7 above)
Comment (klinzo) [reply android/app/buildConfigs.gradle:8]: The lines afterwards can be added here too!

**Review by maxammann — APPROVED** [review]

**Review by maxammann — COMMENTED** [review]
Are you sure this fixed the second problem you had?

The proc.waitFor() is still before the proc.in.text

**Comment by lulo4907** [comment]
@maxammann this will be managed with Native-623

**Pull request merged**